File: android/app/src/main/java/com/scholarlyapps/pathlingo/ui/auth/LogoutHelper.java
package com.scholarlyapps.pathlingo.ui.auth;

import android.content.Context;
import android.content.Intent;

import com.scholarlyapps.pathlingo.data.remote.ServiceLocator;

import java.util.concurrent.Executors;

public class LogoutHelper {

    public static void logout(Context context) {
        ServiceLocator.init(context);
        Executors.newSingleThreadExecutor().execute(() -> ServiceLocator.authRepository.logout());
        Intent intent = new Intent(context, LoginActivity.class)
                .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
        context.startActivity(intent);
    }
}

File: android/app/src/main/java/com/scholarlyapps/pathlingo/ui/auth/LogoutHelper.kt
package com.scholarlyapps.pathlingo.ui.auth

import android.content.Context
import android.content.Intent
import com.scholarlyapps.pathlingo.data.remote.ServiceLocator
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.SupervisorJob
import kotlinx.coroutines.launch

object LogoutHelper {

    private val scope = CoroutineScope(SupervisorJob() + Dispatchers.IO)

    @JvmStatic
    fun logout(context: Context) {
        ServiceLocator.init(context)
        scope.launch { ServiceLocator.authRepository.logout() }
        val intent = Intent(context, AuthActivity::class.java)
            .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
        context.startActivity(intent)
    }
}
